lib/core-server/src/utils/get-manager-builder.ts

```diff
@@ -0,0 +1,14 @@
+import path from 'path';
+import { getInterpretedFile, serverRequire, Options } from '@storybook/core-common';
+
+export async function getManagerBuilder(configDir: Options['configDir']) {
+  const main = path.resolve(configDir, 'main');
+  const mainFile = getInterpretedFile(main);
+  const { core } = mainFile ? serverRequire(mainFile) : { core: null };
+
+  const builderPackage =
+    core?.builder === 'webpack5' ? '@storybook/manager-webpack5' : '@storybook/manager-webpack4';
+
+  const managerBuilder = await import(builderPackage);
+  return managerBuilder;
+}
```

Comment:
What about something like this? it's a bit more scalable if there are going to be more than two builders:

```suggestion
import path from 'path';
import { getInterpretedFile, serverRequire, Options } from '@storybook/core-common';

const builderVersions = {
  webpack4: '@storybook/manager-webpack4',
  webpack5: '@storybook/manager-webpack5',
}

export async function getManagerBuilder(configDir: Options['configDir']) {
  const main = path.resolve(configDir, 'main');
  const mainFile = getInterpretedFile(main);

  if(!mainFile) {
    // use webpack4 by default
    return import(builderVersions.webpack4);
  }
  
  const { core } = serverRequire(mainFile);

  const builderPackage = builderVersions[core.builder];

  const managerBuilder = await import(builderPackage);
  return managerBuilder;
}
```

Reply:
Added clarifying comment in the code!